common/src/update.rs

```diff
@@ -328,6 +328,19 @@ pub fn hex_schema<const N: usize>(gen: &mut SchemaGenerator) -> Schema {
     schema.into()
 }
 
+fn artifact_hash_schema(gen: &mut SchemaGenerator) -> Schema {
```

Comment:
might it be simpler to `impl JsonSchema for ArtifactHash`? And it might make sense to inline hex_schema, minimal as it is.